konveyor/tackle-pathfinder
# TACKLE-300 Spanish version of the questionnaire

Issue : https://github.com/konveyor/tackle-pathfinder/issues/97

Added translation of the questionnaire to Spanish.

In order to test : 

1. clone the code and cd into it
2. execute a pathfinder_db and keycloak

```
podman run -ti --name keycloak --rm \
            -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -e KEYCLOAK_IMPORT=/tmp/keycloak/quarkus-realm.json \
            -e DB_VENDOR=h2 -p 8180:8080 -p 8543:8443 -v ./src/main/resources/keycloak:/tmp/keycloak:Z \
            jboss/keycloak:12.0.2
```
```
podman run -it \                                 
            --name postgres-pathfinder -e POSTGRES_USER=pathfinder \
            -e POSTGRES_PASSWORD=pathfinder -e POSTGRES_DB=pathfinder_db \
            -p 5433:5432 postgres:10.6
```
3. execute the application in dev mode ` mvn quarkus:dev `

4. execute the check_api.sh
```
.github/scripts/check_api.sh localhost:8085 localhost:8180
```
5.  obtain the token
```
export access_token=$(curl -X POST "http://localhost:8180/auth/realms/quarkus/protocol/openid-connect/token" \
            --user backend-service:secret \
            -H 'content-type: application/x-www-form-urlencoded' \
            -d 'username=alice&password=alice&grant_type=password' | jq --raw-output '.access_token')
```

6. obtains the assessment ID
```
curl -X GET "http://localhost:8085/pathfinder/assessments?applicationId=100" -H 'Accept: application/json' \
            -H "Authorization: Bearer $access_token" -s
```
7. obtain assessment, with the id from previous step, in Spanish and check the text is clearly Spanish
```
curl -X GET "http://$api_ip/pathfinder/assessments/{assessmentId}" -H 'Accept: application/json' \
            -H "Authorization: Bearer $access_token"
```
